src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,54 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User is null");
+        }
+        String login = user.getLogin();
+        String password = user.getPassword();
+        Integer age = user.getAge();
+        if (isLoginInvalid(login)) {
+            throw new RegistrationException("Login should be least "
+                    + MIN_LOGIN_LENGTH + "characters");
+        }
+        if (isPasswordInvalid(password)) {
+            throw new RegistrationException("Password should be least "
+                    + MIN_PASSWORD_LENGTH + "characters");
+        }
+        if (isAgeInvalid(age)) {
+            throw new RegistrationException("Age should be least " + MIN_USER_AGE);
+        }
+        if (isLoginExisted(login)) {
+            throw new RegistrationException("Login " + login + " is already used");
+        }
```

Comment:
Cool, but what do you think about extracting all validation logic to the separate method, thus `register` will just call the validation method and proceed next?